### PR TITLE
If an auth token is supplied in config, send it as Bearer

### PR DESCRIPTION
**Pull request by jimmed**
Apologies this is using a personal fork - I don't have any write permissions on the qubitdigital fork.

This is part 2 of https://github.com/qubitdigital/enporium/pull/38

### CODE REVIEW

**Comment by swagatha-christie** [comment]
By analyzing the blame information on this pull request, we identified @fengmk2, @dead-horse and @KidkArolis to be potential reviewers

File: proxy/npm.js [hunk 1]
```diff
@@ -26,6 +26,11 @@ function* request(url, options) {
   options.headers = {
     'user-agent': USER_AGENT
   };
+
+  if (config.auth && config.auth.token) {
```
Comment (KidkArolis) [inline proxy/npm.js]: This looks generally good. I wonder if we should name this config option more similar to the `config.sourceNpmRegistry`? E.g. `config.sourceNpmRegistryToken`

Comment (jimmed) [inline proxy/npm.js]: That makes a lot of sense - I'll do that!

**Comment by rogy** [comment]
legit 👍

**Comment by jimmed** [comment]
@KidkArolis I have no write access, please merge!

main-qimg-c8781a4bb1f17e330b50cb35f851da05

**Pull request merged**